src/v/security/acl_store.h

```diff
@@ -22,6 +24,8 @@ namespace security {
 class acl_entry_set {
 public:
     using container_type = absl::node_hash_set<acl_entry>;
+    using role_cache_type
+      = absl::flat_hash_map<ss::sstring, ssize_t, sstring_hash, sstring_eq>;
```

Comment:
I considered that (it would be better), but it's complicated by one-to-many relationship with ACL set entries. There's no way to track which `acl_entry::principal` initialized the map entry, and it wouldn't necessarily be the last one removed.

Reply:
Fair point!

I also considered `flat_hash_set<lw_shared_ptr<ss::sstring>>`, but that doesn't help!